includes/admin/tasks/class-wc-payments-task-disputes.php

```diff
@@ -125,25 +242,80 @@ public function is_complete() {
 	 * @return bool
 	 */
 	public function can_view() {
-		return $this->get_disputes_awaiting_response_count() > 0;
+		return count( $this->get_disputes_needing_response_within_days( 7 ) ) > 0;
 	}
 
 	/**
-	 * Gets the number of disputes which need a response.
+	 * Get disputes needing response within the given number of days.
 	 *
-	 * Because this task is initialized before WC Payments, we can only fetch from the cache (via "get()").
-	 * The dispute status cache cannot be populated via this task. If this value hasn't been cached yet, this task won't show until it is.
+	 * @param int $num_days Number of days in the future to check for disputes needing response.
 	 *
-	 * @return int The number of disputes which need a response.
+	 * @return array Array of disputes needing response within the given number of days.
 	 */
-	private function get_disputes_awaiting_response_count() {
-		$disputes_status_counts = \WC_Payments::get_database_cache()->get( Database_Cache::DISPUTE_STATUS_COUNTS_KEY );
+	private function get_disputes_needing_response_within_days( $num_days ) {
+		$to_return = [];
+
+		$active_disputes = $this->get_disputes_needing_response();
+		if ( ! $active_disputes ) {
+			return $to_return;
+		}
+
+		foreach ( $active_disputes as $dispute ) {
+			if ( ! $dispute['due_by'] ) {
+				continue;
+			}
 
-		if ( empty( $disputes_status_counts ) ) {
-			return 0;
+			// Compare UTC times.
+			$now_utc    = new \DateTime( 'now', new \DateTimeZone( 'UTC' ) );
+			$due_by_utc = new \DateTime( $dispute['due_by'], new \DateTimeZone( 'UTC' ) );
+
+			if ( $now_utc > $due_by_utc ) {
+				continue;
+			}
+
+			$diff = $now_utc->diff( $due_by_utc );
+			// If the dispute is due within the given number of days, add it to the list.
+			if ( $diff->days <= $num_days ) {
+				$to_return[] = $dispute;
+			}
 		}
 
-		$needs_response_statuses = [ 'needs_response', 'warning_needs_response' ];
-		return (int) array_sum( array_intersect_key( $disputes_status_counts, array_flip( $needs_response_statuses ) ) );
+		return $to_return;
+	}
+
+	/**
+	 * Gets disputes awaiting a response. ie have a 'needs_response' or 'warning_needs_response' status.
+	 *
+	 * @return array|null Array of disputes awaiting a response. Null on failure.
+	 */
+	private function get_disputes_needing_response() {
+		return $this->database_cache->get_or_add(
+			Database_Cache::ACTIVE_DISPUTES_KEY,
+			function() {
+				$response = $this->api_client->get_disputes(
+					[
+						'pagesize' => 50,
+						'search'   => [ 'warning_needs_response', 'needs_response' ],
+					]
+				);
+
+				$active_disputes = $response['data'] ?? [];
+
+				// sort by due_by date ascending.
+				usort(
+					$active_disputes,
+					function( $a, $b ) {
+						$a_due_by = new \DateTime( $a['due_by'] );
+						$b_due_by = new \DateTime( $b['due_by'] );
+
+						return $a_due_by <=> $b_due_by;
```

Comment:
[I tested this](https://3v4l.org/PtkLg) and it can be simplified without the calls to `\DateTime` as the ISO8601 format (without timezone) is lexicographical sorting aligns with chronological.  However, this is probably a micro-optimization and decreases readability.
```suggestion
						return $a['due_by'] <=> $b['due_by'];
```

Reply:
Thanks, I agree this probably helps with readability so I'll leave it.